src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts

```diff
@@ -93,10 +94,17 @@ export class ExerciseDetailsStudentActionsComponent implements OnInit {
                 }
                 return ParticipationStatus.QUIZ_FINISHED;
             }
-        } else if ((this.exercise.type === ExerciseType.MODELING || this.exercise.type === ExerciseType.TEXT) && this.hasParticipations(this.exercise)) {
+        } else if (
+            (this.exercise.type === ExerciseType.MODELING || this.exercise.type === ExerciseType.TEXT || this.exercise.type === ExerciseType.FILE_UPLOAD) &&
+            this.hasParticipations(this.exercise)
+        ) {
             const participation = this.exercise.participations[0];
             if (participation.initializationState === InitializationState.INITIALIZED || participation.initializationState === InitializationState.FINISHED) {
-                return this.exercise.type === ExerciseType.MODELING ? ParticipationStatus.MODELING_EXERCISE : ParticipationStatus.TEXT_EXERCISE;
+                return this.exercise.type === ExerciseType.MODELING
```

Comment:
I think
```suggestion
                switch (this.exercise.type) {
                    case ExerciseType.MODELING: return ParticipationStatus.MODELING_EXERCISE;
                    case ExerciseType.TEXT: return ParticipationStatus.TEXT_EXERCISE;
                    default: return ParticipationStatus.FILE_UPLOAD_EXERCISE;
                }
```
is more readable.